core/src/main/scala/kafka/server/MetadataCache.scala

```diff
@@ -59,12 +60,15 @@ class MetadataCache(brokerId: Int) extends Logging {
   // filterUnavailableEndpoints exists to support v0 MetadataResponses
   private def getEndpoints(snapshot: MetadataSnapshot, brokers: Iterable[java.lang.Integer], listenerName: ListenerName, filterUnavailableEndpoints: Boolean): Seq[Node] = {
     val result = new mutable.ArrayBuffer[Node](math.min(snapshot.aliveBrokers.size, brokers.size))
-    brokers.foreach { brokerId =>
-      val endpoint = getAliveEndpoint(snapshot, brokerId, listenerName) match {
-        case None => if (!filterUnavailableEndpoints) Some(new Node(brokerId, "", -1)) else None
-        case Some(node) => Some(node)
+    for (brokerId <- brokers) {
+      val endpoint = getAliveEndpoint(snapshot, brokerId, listenerName)
+      if (endpoint != null && endpoint.isDefined) {
```

Comment:
endpoint has the type Option, thus I think this block doesn't involve the java stream API. Am I missing something?

If it turns out these changes are required, then I wonder whether you see null being returned for endpoint? If not, I think there is no need to check against null, as in the original code.

Reply:
@gitlw I had made a code change previously to `getAliveEndpoint` which was incorrect and returning null. So, I didn't include that change. 
You are right. Since it is returning an `Option`, we don't need the null check. I will fix this part. Thanks!